mypyc/test/test_struct.py

```diff
@@ -0,0 +1,104 @@
+import unittest
+
+from mypyc.ir.rtypes import (
+    RStruct, bool_rprimitive, int64_rprimitive, int32_rprimitive, object_rprimitive, StructInfo,
+    int_rprimitive, short_int_rprimitive
+)
+from mypyc.common import IS_32_BIT_PLATFORM
+from mypyc.rt_subtype import is_runtime_subtype
+
+
+class TestStruct(unittest.TestCase):
+    def test_struct_offsets(self) -> None:
+        # test per-member alignment
+        info = StructInfo("", [], [bool_rprimitive, int32_rprimitive, int64_rprimitive])
+        r = RStruct(info)
+        assert r.size == 16
+        assert r.offsets == [0, 4, 8]
+
+        # test final alignment
+        info1 = StructInfo("", [], [bool_rprimitive, bool_rprimitive])
+        r1 = RStruct(info1)
+        assert r1.size == 2
+        assert r1.offsets == [0, 1]
+        info2 = StructInfo("", [], [int32_rprimitive, bool_rprimitive])
+        r2 = RStruct(info2)
+        info3 = StructInfo("", [], [int64_rprimitive, bool_rprimitive])
+        r3 = RStruct(info3)
+        assert r2.offsets == [0, 4]
+        assert r3.offsets == [0, 8]
+        if IS_32_BIT_PLATFORM:
+            assert r2.size == 8
+            assert r3.size == 12
+        else:
+            assert r2.size == 8
+            assert r3.size == 16
+
+        info4 = StructInfo("", [], [bool_rprimitive, bool_rprimitive,
+                              bool_rprimitive, int32_rprimitive])
+        r4 = RStruct(info4)
+        assert r4.size == 8
+        assert r4.offsets == [0, 1, 2, 4]
+
+        # test nested struct
+        info5 = StructInfo("", [], [bool_rprimitive, r])
+        r5 = RStruct(info5)
+        assert r5.offsets == [0, 8]
+        assert r5.size == 24
+        info6 = StructInfo("", [], [int32_rprimitive, r5])
+        r6 = RStruct(info6)
+        assert r6.offsets == [0, 8]
+        assert r6.size == 32
+
+    def test_struct_str(self) -> None:
+        info = StructInfo("Foo", ["a", "b"],
+                    [bool_rprimitive, object_rprimitive])
+        r = RStruct(info)
+        assert str(r) == "Foo{a:bool, b:object}"
+        assert repr(r) == "<RStruct Foo{a:<RPrimitive builtins.bool>, " \
+                          "b:<RPrimitive builtins.object>}>"
+        info1 = StructInfo("Bar", ["c"], [int32_rprimitive])
+        r1 = RStruct(info1)
+        assert str(r1) == "Bar{c:int32}"
+        assert repr(r1) == "<RStruct Bar{c:<RPrimitive int32>}>"
+        info2 = StructInfo("Baz", [], [])
+        r2 = RStruct(info2)
+        assert str(r2) == "Baz{}"
+        assert repr(r2) == "<RStruct Baz{}>"
+
+    def test_runtime_subtype(self) -> None:
+        # right type to check with
+        info = StructInfo("Foo", ["a", "b"],
+                    [bool_rprimitive, int_rprimitive])
+        r = RStruct(info)
+
+        # second type is subtype of r's second type
+        info1 = StructInfo("Foo", ["a", "b"],
+                    [bool_rprimitive, short_int_rprimitive])
```

Comment:
Since we plan to ensure that for each name there will only be a single struct, this doesn't make sense as a test case. It's sufficient to test that a struct is a runtime subtype only of the same struct.